### ISSUE TITLE
[import/no-unresolved] <issue title>

### ISSUE THREAD

**Issue by jjangga0214** [typescript-eslint/typescript-eslint#991]
# Repro

##### tree

```
<root>
├── src
│   ├── foo.ts
│   └── index.ts
├── tsconfig.json
├── .eslintrc.js
...
└── package.json
```

##### tsconfig.json

Other options are omiited for brevity.

```js
{
  "compilerOptions": {
    "baseUrl": "./src"
    "paths": {
      "#/*": ["./*"]
    }
  }
}
```
##### .eslintrc.js

Other options are omiited for brevity.

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  root: true,
  env: {
    node: true,
    es6: true,
  },
  plugins: ['@typescript-eslint', ],
  extends: [
    'plugin:@typescript-eslint/recommended',
    'plugin:import/typescript',
  ],
}

```

##### src/index.ts

```ts
import { foo } from '#/foo'

console.log(`index.ts uses ${foo}`)
```

# Problem

**Expected Result**

It should respect `baseUrl` and `paths` option on tsconfig.json

**Actual Result**

```
error  Unable to resolve path to module '#/foo'  import/no-unresolved
```

**Additional Info**

<https://github.com/jjangga0214/ts-boilerplate/commit/af74c178a3c5ee84f78c520ba3c09ad73a9fc130/checks> is the actual result with an error message. (Expand it by clicking "Lint" on github action checks) 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.2.0` |
| `@typescript-eslint/parser`        | `2.2.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `5.3.0` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.10.2` |
| `yarn`                              | `1.17.3` |

**Comment by bradzacher** [comment]
We do not have any association with `eslint-plugin-import`, and cannot fix bugs with it.
You should consider raising an issue in either that repo, or `eslint-import-resolver-typescript`
